yt/frontends/amrvac/fields.py

```diff
@@ -29,14 +29,43 @@
     "spherical": ("r", "theta", "phi")
 }
 
+def _velocity(idir, idust=None):
+    """Generate a velocity function v_idir = m_idir / rho
+
+    If idust is not used, the "gas" velocity function is returned.
+
+    Parameters
+    ----------
+    idir : int
+        the direction index (1, 2 or 3)
+    idust : int, optional
+        the dust species index (AMRVAC). idust >= 1
+    """
+    if idust is None:
+        dust_label = ""
+    else:
+        dust_label = "dust%d_" % idust
+    def velocity_idir(field, data):
+        return data["gas", "%smoment_%d" % (dust_label, idir)] / data["gas", "%sdensity" % dust_label]
+    return velocity_idir
```

Comment:
I'm having a hard time actually 'reading' this (never used dust before in amrvac), can you put some notes here what it should be when dust is present? I'm reading the fields now as `('gas', 'dust1_moment_1')`, where it's possible to have this up to `dust100_moment_1`, correct?